packages/common-all/src/dnode.ts

```diff
@@ -1218,48 +1218,43 @@ export class NoteUtils {
    * @param opts.includeSelf: If true, note with `fname` itself will be included in the ancestors.
    * @param opts.nonStubOnly: If true, only notes that are not stubs will be included.
    */
-  static *ancestors(opts: {
+  static async ancestors(opts: {
     fname: string;
     vault?: DVault;
     engine: DEngineClient;
     includeSelf?: boolean;
     nonStubOnly?: boolean;
-  }): Generator<NoteProps> {
+  }): Promise<NotePropsMeta | undefined> {
```

Comment:
FWIW, there's an AsyncGenerator that also works here, but if the new signature works then that's also fine. 

```typescript
  static async *ancestors(opts: {
    fname: string;
    vault?: DVault;
    engine: DEngineClient;
    includeSelf?: boolean;
    nonStubOnly?: boolean;
  }): AsyncGenerator<NoteProps> {
```

Reply:
Got it. I'm not sure if generator is needed here. The only use case I see is for the `color` method which it seems you've commented out for now lol